Svc/SeqDispatcher/SeqDispatcher.cpp

```diff
@@ -0,0 +1,210 @@
+// ======================================================================
+// \title  SeqDispatcher.cpp
+// \author zimri.leisher
+// \brief  cpp file for SeqDispatcher component implementation class
+// ======================================================================
+
+#include <Svc/SeqDispatcher/SeqDispatcher.hpp>
+
+namespace Svc {
+
+// ----------------------------------------------------------------------
+// Construction, initialization, and destruction
+// ----------------------------------------------------------------------
+
+SeqDispatcher ::SeqDispatcher(const char* const compName)
+    : SeqDispatcherComponentBase(compName) {}
+
+SeqDispatcher ::~SeqDispatcher() {}
+
+void SeqDispatcher::init(NATIVE_INT_TYPE queueDepth,
+                         NATIVE_INT_TYPE instance) {
+  SeqDispatcherComponentBase::init(queueDepth, instance);
+  for (FwIndexType i = 0; i < SeqDispatcherSequencerPorts; i++) {
```

Comment:
I mentioned earlier that this function might be removed. However, this is a fine use of the function: no additional inputs.

I do wonder why this asserts.  Could it be better to just build up a list of sequencers that are available and use those as the pool?

This architecture Implies that every project must attach all command sequencers possible to this component.  You couldn't say configure a sequencer pool max size of 6, but only attach 3.

Reply:
Looking through the whole code, I recommend this check and the whole function be deleted.  See reasons below.

Please feel to comment back if you disagree with my rationale as we should try to reach a consensus.